posthog/hogql/transforms/test/__snapshots__/test_in_cohort.ambr

```diff
@@ -1,4 +1,106 @@
 # serializer version: 1
+# name: TestInCohort.test_in_cohort_conjoined_dynamic
+  '''
+  -- ClickHouse
+  
+  SELECT events.event AS event 
+  FROM events LEFT JOIN (
+  SELECT person_static_cohort.person_id AS cohort_person_id, 1 AS matched, person_static_cohort.cohort_id AS cohort_id 
+  FROM person_static_cohort 
+  WHERE and(equals(person_static_cohort.team_id, 420), in(person_static_cohort.cohort_id, [1])) UNION ALL 
+  SELECT cohortpeople.person_id AS cohort_person_id, 1 AS matched, cohortpeople.cohort_id AS cohort_id 
+  FROM cohortpeople 
+  WHERE and(equals(cohortpeople.team_id, 420), in(cohortpeople.cohort_id, [1])) 
+  GROUP BY cohort_person_id, cohortpeople.cohort_id, cohortpeople.version 
+  HAVING ifNull(greater(sum(cohortpeople.sign), 0), 0)) AS __in_cohort ON equals(__in_cohort.cohort_person_id, events.person_id) 
```

Comment:
flyby: cohortpeople.sign is deprecated, shouldn't be used, this doesn't work properly here. Rather, we look at the version, if it's equal to the version on the cohort.

I see sign is still used in a few places for cohort calculations, but I think those places don't face this same issue 🙈 (for various reasons). In querying though, stick to version!

This means _somehow_ passing the current cohort version value to the query generator

Reply:
basically, the problem here is some stale versions of the cohorts might have sum(sign) > 0 as well, which can then lead to extra persons being filtered, iirc.

Reply:
@neilkakkar Cool, so, I need to grab `cohort.version` and filter `cohort_people` + removing the `sign` clause? 👌

Reply:
Yep, should work well!

This means you can get rid of the group by too, which should speed things up a lot, just copy this query:

https://github.com/PostHog/posthog/blob/master/posthog/models/cohort/sql.py#L75

Reply:
Dreamy, have added this in, but need to remove the group by. Thanks!